numpy/core/src/multiarray/item_selection.c

```diff
@@ -2128,6 +2129,69 @@ count_nonzero_bytes_384(const npy_uint64 * w)
     return r;
 }
 
+#if NPY_SIMD
+
+/* Count the zero bytes between `*d` and `end`, updating `*d` to point to where to keep counting from. */
+static NPY_INLINE NPY_GCC_OPT_3 npyv_u8
+count_zero_bytes_u8(const npy_uint8 **d, const npy_uint8 *end)
+{
+    const npyv_u8 vone = npyv_setall_u8(1);
+    const npyv_u8 vzero = npyv_zero_u8();
+
+    npy_intp lane_max = 0;
+    npyv_u8 vsum8 = npyv_zero_u8();
+    while (*d < end && lane_max <= 0xFE) {
```

Comment:
```suggestion
    while (*d < end && lane_max <= 0xFF - 1) {
```

Reply:
Why not `< 0xFF`? Is `<=` faster?

Reply:
Because the intuition is that the counter should reach at most `0xFF`, and is incremented by one each loop. Writing it this way makes it generalize well to the u16 and u32 cases.